core/src/data_sources/data_source.rs

```diff
@@ -750,22 +752,42 @@ impl DataSource {
             .await?;
         }
 
-        // Upsert document (SQL).
-        store
-            .upsert_data_source_document(
-                &self.project,
-                &self.data_source_id,
-                &main_collection_document,
-                title,
-                mime_type,
-            )
+        // Store upsert does not save the text, token count or chunks.
+        // These fields don't actually exist in the SQL table.
+        // Because of this, we have to manually construct the UpsertDocument, and save
+        // owned values for text and token count so we can return them.
+        // We also don't have type safety as we just use default/empty values for chunks
+        // for code paths where we don't need / have them.
+        // TODO(@fontanierh): use a different type for "DocumentWithTextAndTokenCount"
+        let doc_text = main_collection_document.text;
+        let doc_token_count = main_collection_document.token_count;
+        let create_params = DocumentCreateParams {
+            document_id: main_collection_document.document_id,
+            title: Some(main_collection_document.title),
+            mime_type: Some(main_collection_document.mime_type),
+            timestamp: main_collection_document.timestamp,
+            tags: main_collection_document.tags,
+            parents: main_collection_document.parents,
+            source_url: main_collection_document.source_url,
+            hash: main_collection_document.hash,
+            text_size: main_collection_document.text_size,
+            chunk_count: main_collection_document.chunk_count,
+            created,
```

Comment:
created: main_collection_document.created should work no? that's what we used to do?